src/clib/lib/config/config_parser.cpp

```diff
@@ -565,31 +576,53 @@ config_parse__(config_parser_type *config, config_content_type *content,
 
             // Add define
             else if (define_kw && (strcmp(define_kw, kw) == 0)) {
-                if (active_tokens < 3)
-                    util_abort("%s: keyword:%s must have exactly one (or more) "
-                               "arguments. \n",
-                               __func__, define_kw);
-
-                char *key = (char *)util_alloc_string_copy(
-                    stringlist_iget(token_list, 1));
-                char *value = stringlist_alloc_joined_substring(
-                    token_list, 2, active_tokens, " ");
+                if (active_tokens < 3) {
+                    content->parse_errors.push_back(fmt::format(
+                        "Keyword:{} must have two or more arguments.",
+                        define_kw));
+                    return;
+                }
 
-                config_content_add_define(content, key, value);
+                const std::string key = stringlist_iget(token_list.get(), 1);
+                const int argc = stringlist_get_size(token_list.get());
+                for (int iarg = 2; iarg < argc; iarg++) {
+                    int env_offset = 0;
+                    while (true) {
+                        auto env_var =
+                            char_to_optstr(res_env_isscanf_alloc_envvar(
+                                stringlist_iget(token_list.get(), iarg),
+                                env_offset));
+                        if (!env_var.has_value())
+                            break;
+
+                        const char *env_value =
+                            getenv(env_var.value().substr(1).c_str());
+                        if (env_value != nullptr) {
+                            char *new_value = util_string_replace_alloc(
```

Comment:
This one will need to be freed also? Do you even need to allocate for this?

Reply:
Fixed this with use of unique_ptr